### ISSUE TITLE
Add support for set literal

### ISSUE THREAD

**Issue by xolott** [hashicorp/terraform-provider-helm#1158]
### Description

When I deploy a helm chart using a JSON as a value, helm tries to parse the JSON as a template string and fails with something like ` key "\n        \"enable\": true" has no value (cannot end with ,)`

Helm 3.12.0 added `--set-literal`

Can we add this option to the set block?

### Potential Terraform Configuration

```hcl
resource "helm_release" "my_app" {
  name       = "app"
  repository = "https://my.app.io/helm-charts/"
  chart      = "my-app-chart"

  set {
    name = "my.json.setting"
    type  =  "literal"
    value =  file("${path.module}/my-file.json")
  }
}
```

Or with a `set_literal` block

```hcl
resource "helm_release" "my_app" {
  name       = "app"
  repository = "https://my.app.io/helm-charts/"
  chart      = "my-app-chart"

  set_literal {
    name = "my.json.setting"
    value =  file("${path.module}/my-file.json")
  }
}
```

### References

- [Helm release](https://github.com/helm/helm/releases/tag/v3.12.0)
- [Feature commit](https://github.com/helm/helm/commit/451603910e29ce732e81c2ef5c2f45d252c07a26)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by fabrice-toussaint** [comment]
I agree, Helm already released a fix for this issue: https://github.com/helm/helm/pull/9182

**Comment by xolott** [comment]
We got reactions and no comments. Let's keep this issue alive

**Comment by iBrandyJackson** [comment]
Issue has been addressed and solved with https://github.com/hashicorp/terraform-provider-helm/pull/1615 which is now LIVE in the latest Helm version. 

Closing as complete, please open a new issue if this indeed does not solve this. Thanks!
